### PR TITLE
Enable declare_strict_types option and apply @PHP80Migration set with…

### PR DESCRIPTION
**Pull request by sidz**
This PR enables `declare_strict_types` option and applies `@PHP80Migration` set with risky rules.

### CODE REVIEW

File: src/AbstractUpdateAction.php [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
```
Comment (ruflin) [inline src/AbstractUpdateAction.php:3]: Is this line automatically added to each file or is this something done manually?

**Review by ruflin — COMMENTED** [review]
Looks like this triggered quite a list of CI failures :-)

**Comment by sidz** [comment]
> Looks like this triggered quite a list of CI failures :-)

Yeah. Here is a lot what need to be done.

File: src/AbstractUpdateAction.php (hunk 1 above)
Comment (sidz) [reply src/AbstractUpdateAction.php:3]: With the help of PHP-CS-Fixer `declare_strict_types` rule included in `@PHP80Migration:risky` rule set.

Comment (ruflin) [reply src/AbstractUpdateAction.php:3]: Great, I was worried there would a be manual step involved.

**Comment by sidz** [comment]
ready for review @ruflin

File: tests/Cluster/SettingsTest.php [hunk 2]
```diff
@@ -38,7 +40,7 @@ public function testSetReadOnly(): void
             $index->addDocument($doc2);
             $this->fail('should throw read only exception');
         } catch (ClientResponseException $e) {
-            $error = \json_decode($e->getResponse()->getBody(), true)['error'] ?? null;
+            $error = \json_decode((string) $e->getResponse()->getBody(), true)['error']['root_cause'][0] ?? null;
```
Comment (ruflin) [inline tests/Cluster/SettingsTest.php:43]: Do you know why this changed?

File: tests/Index/SettingsTest.php [hunk 3]
```diff
@@ -136,8 +137,7 @@ public function testDeleteAliasWithException(): void
             $indexAlias->delete();
             $this->fail('Should throw exception because you should delete the concrete index and not the alias');
         } catch (ClientResponseException $e) {
-            $response = ResponseConverter::toElastica($e->getResponse());
-            $error = $response->getFullError();
+            $error = \json_decode((string) $e->getResponse()->getBody(), true)['error']['root_cause'][0] ?? null;
```
Comment (ruflin) [inline tests/Index/SettingsTest.php:140]: What is the reason you removed the ResponseCoverter here?

File: tests/Query/CombinedFieldsQueryTest.php [hunk 4]
```diff
@@ -16,10 +18,10 @@
 class CombinedFieldsQueryTest extends BaseTest
 {
     private static $data = [
-        ['id' => 1, 'title' => 'Rodolfo', 'body' => 'Moraes',   'abstract' => 'Lorem'],
-        ['id' => 2, 'title' => 'Tristan', 'body' => 'Maindron', 'abstract' => 'Dolor'],
-        ['id' => 3, 'title' => 'Monique', 'body' => 'Maindron', 'abstract' => 'Ipsum'],
-        ['id' => 4, 'title' => 'John',    'body' => 'not Doe',  'abstract' => 'Consectetur'],
+        ['id' => '1', 'title' => 'Rodolfo', 'body' => 'Moraes',   'abstract' => 'Lorem'],
```
Comment (ruflin) [inline tests/Query/CombinedFieldsQueryTest.php:21]: I assume because of line 148, the old syntax would still work?

**Review by ruflin — APPROVED** [review]
Change LGTM. I'll get this in directly to make sure we don't have any future conflicts. I have one question around a change in the test but we can still follow up on this separately.

**Comment by ruflin** [comment]
Thanks for all the cleanup @sidz One thing I just noticed after merging, no changelog 🤦

File: tests/Index/SettingsTest.php (hunk 3 above)
Comment (sidz) [reply tests/Index/SettingsTest.php:140]: `ResponseCoverter` in my opinion should work with code related to `src` folder  (production code). Official Elasticsearch client does not return `ResponseInterface`.

I'm planning to raise a new PR where we won't allow ResponseInterface as an option to `ResponseCoverter`

File: tests/Query/CombinedFieldsQueryTest.php (hunk 4 above)
Comment (sidz) [reply tests/Query/CombinedFieldsQueryTest.php:21]: Document id must be string or null. see https://github.com/ruflin/Elastica/blob/8.x/src/Document.php#L45

And it was added long time ago. Clients code must respect this and pass string or null only.

**Comment by sidz** [comment]
> Thanks for all the cleanup @sidz One thing I just noticed after merging, no changelog 🤦

I'll add in the separate PR

File: tests/Cluster/SettingsTest.php (hunk 2 above)
Comment (sidz) [reply tests/Cluster/SettingsTest.php:43]: that's a good catch:
Seems like elasticsearch response looks like:
```
{
  "error": {
    "type": "cluster_block_exception",
    "root_cause": [
      {
        "type": "cluster_block_exception",
        ...
      }
    ]
  }
}
```

Comment (ruflin) [reply tests/Cluster/SettingsTest.php:43]: I wonder why the test passed before 🤔  If I remember correctly, there was a major ES version where the response format changed, but that was quite a while ago. Good to know we have it covered in the tests.

File: tests/Index/SettingsTest.php (hunk 3 above)
Comment (ruflin) [reply tests/Index/SettingsTest.php:140]: Does this make migration from 7.x potentially harder? Lets discuss on the upcoming PR, code will make it much easier to fully understand what you mean by this.

**Pull request merged**